lib/builtins/deploy-delegates/cfn-deployer/helper.js

```diff
@@ -133,6 +137,18 @@ function deployStack(reporter, options, callback) {
     });
 }
 
+function _mergeTemplateParameters(stackParameters, templateParameters = {}) {
+    Object.keys(templateParameters).forEach(ParameterKey => {
+        const ParameterValue = templateParameters[ParameterKey];
+        const index = stackParameters.findIndex(p => p.ParameterKey === ParameterKey);
+        if (index > -1) {
+            stackParameters[index].ParameterValue = ParameterValue;
+        } else {
+            stackParameters.push({ ParameterKey, ParameterValue });
+        }
+    });
+}
```

Comment:
Can we use https://ramdajs.com/docs/#mergeDeepLeft to do the (deep) merge?

Reply:
merge deep left merges on property keys. we are merging arrays. Tried and it did not work as expected:
```
R.mergeDeepLeft([{ParameterKey: 'z', ParameterValue: 'f'}, {ParameterKey: 'x', ParameterValue: 'y'}],
[{ParameterKey: 'z', ParameterValue: 'r'}, {ParameterKey: 'q', ParameterValue: 'y'}]);
```